### PR TITLE
fix: next array

### PR DESCRIPTION
**Pull request by sattvikc**
## Summary of change

(A few sentences about this PR)

## Related issues

-   Link to issue1 here
-   Link to issue1 here

## Test Plan

(Write your test plan here. If you changed any code, please provide us with clear instructions on how you verified your changes work. Bonus points for screenshots and videos!)

## Documentation changes

(If relevant, please create a PR in our [docs repo](https://github.com/supertokens/docs), or create a checklist here highlighting the necessary changes)

## Checklist for important updates

-   [ ] Changelog has been updated
-   [ ] `coreDriverInterfaceSupported.json` file has been updated (if needed)
    -   Along with the associated array in `lib/ts/version.ts`
-   [ ] `frontendDriverInterfaceSupported.json` file has been updated (if needed)
-   [ ] Changes to the version if needed
    -   In `package.json`
    -   In `package-lock.json`
    -   In `lib/ts/version.ts`
-   [ ] Had run `npm run build-pretty`
-   [ ] Had installed and ran the pre-commit hook
-   [ ] If new thirdparty provider is added,
    -   [ ] update switch statement in `recipe/thirdparty/providers/configUtils.ts` file, `createProvider` function.
    -   [ ] add an icon on the user management dashboard.
-   [ ] Issue this PR against the latest non released version branch.
    -   To know which one it is, run find the latest released tag (`git tag`) in the format `vX.Y.Z`, and then find the latest branch (`git branch --all`) whose `X.Y` is greater than the latest released tag.
    -   If no such branch exists, then create one from the latest released branch.
-   [ ] If have added a new web framework, update the `add-ts-no-check.js` file to include that
-   [ ] If added a new recipe / api interface, then make sure that the implementation of it uses NON arrow functions only (like `someFunc: function () {..}`).
-   [ ] If added a new recipe, then make sure to expose it inside the recipe folder present in the root of this repo. We also need to expose its types.

## Remaining TODOs for this PR

-   [ ] Item1
-   [ ] Item2

### CODE REVIEW

File: lib/ts/recipe/multifactorauth/api/implementation.ts [hunk 1]
```diff
@@ -21,7 +21,7 @@ import SessionError from "../../session/error";
 
 export default function getAPIInterface(): APIInterface {
     return {
-        updateSessionAndFetchMfaInfoPUT: async ({ options, session, userContext }) => {
+        resyncSessionAndFetchMFAInfoPUT: async ({ options, session, userContext }) => {
```
Comment (porcellus) [inline lib/ts/recipe/multifactorauth/api/implementation.ts:24]: Why was this renamed? I'm not completely against it but I like update a bit better because it matches with `updateAccessTokenPayload`.

File: lib/ts/recipe/multifactorauth/multiFactorAuthClaim.ts [hunk 2]
```diff
@@ -143,7 +133,37 @@ export class MultiFactorAuthClaimClass extends SessionClaim<MFAClaimValue> {
         hasCompletedFactors(requirements: MFARequirementList, id?: string): SessionClaimValidator;
     };
 
-    public buildNextArray(completedClaims: MFAClaimValue["c"], requirements: MFARequirementList): string[] {
+    public isRequirementsSatisfied(completedClaims: MFAClaimValue["c"], requirements: MFARequirementList): boolean {
```
Comment (porcellus) [inline lib/ts/recipe/multifactorauth/multiFactorAuthClaim.ts:136]:
```suggestion
    public isRequirementListSatisfied(completedClaims: MFAClaimValue["c"], requirementList: MFARequirementList): boolean {
```

Comment (porcellus) [reply lib/ts/recipe/multifactorauth/multiFactorAuthClaim.ts:136]: also, this partially duplicates the functionality of `getNextSetOfUnsatisfiedFactors`. I think we should de-duplicate this code (this is kind of the same as checking if the length of unsatisfied factors is 0)

**Review by porcellus — CHANGES_REQUESTED** [review]

File: lib/ts/recipe/multifactorauth/multiFactorAuthClaim.ts (hunk 2 above)
Comment (sattvikc) [reply lib/ts/recipe/multifactorauth/multiFactorAuthClaim.ts]: Done

File: lib/ts/recipe/multifactorauth/multiFactorAuthClaim.ts [hunk 3]
```diff
@@ -47,20 +47,10 @@ export class MultiFactorAuthClaimClass extends SessionClaim<MFAClaimValue> {
...
-                    const { n } = claimVal;
-
-                    if (n.length === 0) {
-                        return {
-                            isValid: true,
-                        };
-                    }
+                    const { v } = claimVal;
 
                     return {
```
[the remark this answers is not in the document]
Comment (sattvikc) [reply lib/ts/recipe/multifactorauth/multiFactorAuthClaim.ts:52]: Done
